File: polyglot/app_yaml_static_config/py/src/app_yaml_static_config/validators.py
from typing import Any

class ConfigurationError(Exception):
    def __init__(self, message: str, context: Any = None):
        self.context = context
        super().__init__(f"{message} (context: {context})")

class ImmutabilityError(ConfigurationError):
    pass

def validate_config_key(key: str) -> None:
    if not key:
        raise ConfigurationError("Config key cannot be empty")

File: polyglot/app_yaml_static_config/mjs/src/validators.ts
export class ConfigurationError extends Error {
    constructor(message: string, public context?: unknown) {
        super(`${message} (context: ${JSON.stringify(context)})`);
        this.name = 'ConfigurationError';
    }
}

export class ImmutabilityError extends ConfigurationError {
    constructor(message: string, context?: unknown) {
        super(message, context);
        this.name = 'ImmutabilityError';
    }
}

export function validateConfigKey(key: string): void {
    if (!key) {
        throw new ConfigurationError("Config key cannot be empty");
    }
}
